### ISSUE TITLE
'JSON and serialization' page issue

### ISSUE THREAD

**Issue by richardhaven** [flutter/website#5182]
Page URL: https://flutter.dev/docs/development/data-and-backend/json.html
Page source: https://github.com/flutter/website/tree/master/src/docs/development/data-and-backend/json.md

Found a typo? You can fix it yourself by going to the page source and clicking the pencil icon. Or finish creating this issue.

Description of issue: "Encoding and serialization are the same thing—turning a data structure into a string"

I understand some simplification and generalization, but this is flat-out WRONG. Please make the context of this specific example clear

**Comment by sfshaza2** [comment]
@johnpryan, @domesticmouse, can we clarify this?

**Comment by domesticmouse** [comment]
Apologies for going all citation required, but can you please explain why you believe that Encoding and serialization are different concepts?

According to [Wikipedia's Code page](https://en.wikipedia.org/wiki/Code): The process of encoding converts information from a source into symbols for communication or storage.

Likewise, the [Wikipedia page on Serialization](https://en.wikipedia.org/wiki/Serialization) states: In computing, serialization (US spelling) or serialisation (UK spelling) is the process of translating a data structure or object state into a format that can be stored (for example, in a file or memory data buffer) or transmitted (for example, across a computer network) and reconstructed later (possibly in a different computer environment).

I'm having trouble figuring the difference between the two definitions. Please enlighten me?

**Comment by richardhaven** [comment]
One can encode something (e.g. UTF-16, xor) without creating something to be transmitted and received

I grant that the difference is small  in practice, but the two terms have different intents, and perhaps that difference deserves two different words

Cheers

**Comment by domesticmouse** [comment]
I totally agree it would be good to have two different words for these closely related concepts, but we have to deal with words as they are currently used in practice. However, given the currently accepted definitions for the words, I'd argue the statement "Encoding and serialization are the same thing—turning a data structure into a string" is more true than false, as much as we might like it to be otherwise.

**Comment by richardhaven** [comment]
One can encode a string into a non-string (e.g. zip files)

Cheers
